src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -295,32 +314,40 @@ private void HandleTableKindNormal(SearchParamTableExpression searchParamTableEx
                 // if this is not sort mode or if it is the first cte
                 if (!IsInSortMode(context) || predecessorIndex < 0)
                 {
-                    StringBuilder.Append("SELECT ").Append(VLatest.Resource.ResourceSurrogateId, null).AppendLine(" AS Sid1")
+                    StringBuilder.Append("SELECT ")
+                        .Append(VLatest.Resource.ResourceTypeId, null).Append(" AS T1, ")
+                        .Append(VLatest.Resource.ResourceSurrogateId, null).AppendLine(" AS Sid1")
                         .Append("FROM ").AppendLine(searchParamTableExpression.QueryGenerator.Table);
                 }
                 else
                 {
                     // we are in sort mode and we need to join with previous cte to propagate the SortValue
                     var cte = TableExpressionName(predecessorIndex);
-                    StringBuilder.Append("SELECT ").Append(VLatest.Resource.ResourceSurrogateId, null).Append(" AS Sid1, ")
+                    StringBuilder.Append("SELECT ")
+                        .Append(VLatest.Resource.ResourceTypeId, null).Append(" AS T1, ")
+                        .Append(VLatest.Resource.ResourceSurrogateId, null).Append(" AS Sid1, ")
                         .Append(cte).AppendLine(".SortValue")
                         .Append("FROM ").AppendLine(searchParamTableExpression.QueryGenerator.Table)
                         .Append("INNER JOIN ").AppendLine(cte);
 
                     using (var delimited = StringBuilder.BeginDelimitedOnClause())
                     {
+                        delimited.BeginDelimitedElement().Append(VLatest.Resource.ResourceTypeId, null).Append(" = ").Append(cte).Append(".T1");
                         delimited.BeginDelimitedElement().Append(VLatest.Resource.ResourceSurrogateId, null).Append(" = ").Append(cte).Append(".Sid1");
                     }
                 }
             }
             else
             {
-                StringBuilder.Append("SELECT Sid1, ").Append(VLatest.Resource.ResourceSurrogateId, null).AppendLine(" AS Sid2")
+                StringBuilder.Append("SELECT T1, Sid1, ")
+                    .Append(VLatest.Resource.ResourceTypeId, null).AppendLine(" AS T2, ")
```

Comment:
Just for my understanding - now we are generating queries where we are selecting the ResourceTypeId in all cases. Will this have a performance impact on tables that are using schemas older than `SchemaVersionConstants.PartitionedTables`?

Reply:
`ResourceTypeId` was always an included column in nonclustered indexes, and we were already using in `WHERE` clauses, so these new queries should have very similar performance.